source/common/termination_predicate_impl.cc

```diff
@@ -16,8 +16,8 @@ TerminationPredicate::Status TerminationPredicateBaseImpl::evaluateChain() {
 }
 
 TerminationPredicate::Status DurationTerminationPredicateImpl::evaluate() {
-  return time_source_.monotonicTime() - start_ > duration_ ? TerminationPredicate::Status::TERMINATE
-                                                           : TerminationPredicate::Status::PROCEED;
+  return time_source_.systemTime() - start_ > duration_ ? TerminationPredicate::Status::TERMINATE
```

Comment:
Well, `SystemTime` doesn't guarantee to always move forward across calls to get snapshots of it, like `MonotonicTime` does, and may it be adjusted while we are polling it. But there's only a very small window in which this can affect operation here: the duration that the main thread requests workers to wait before starting execution. That delay is computed here:
https://github.com/envoyproxy/nighthawk/blob/6aa03318440805719d4953fbbaa3c4d16c433b39/source/client/process_impl.cc#L167
 
Reasoning through clock updates that get applied right between our scheduling and starting of operations: 
1. with small updates, load generation may start a little earlier or later, no problem. Any durations that get measured for latency or execution are based on monotonic time and will not be affected.
2. when the clock jumps forward a lot, worst case workers won't have sufficient time to get ready to start because the clock moved back in time significantly, but they will observe that and complain in the logs about it. (Execution results may be noisy because of workers having missed their schedules to start).
3. when the clock jumps backwards a lot, workers will wait longer before starting execution. This isn't a problem, unless it's a huge leap backwards in time, in which case the wait might take a long time as well. 

Also, suspend/sleep might work a little differently, I suspect that `MonotonicTime` may not track time spend suspended/sleeping.  `3.` from above more or less applies here as well.

All in all, I think chances are pretty small of anyone running into trouble because of this?